Upgrade maps to v23

diegotori opened the pull request:
Upgraded Google API Maps to version 23, revision 1 in "scripts/install-dependencies.sh" (as a follow-up to pull request #2248). Also kept trailing version of previous Maps version just in case (feel free to ask for its removal if necessary). Finally, TravisCI config is now fetching "addon-google_apis-google-23" and "android-23" as a result of these changes.

jongerrish commented:
Can you merge please?

I think you may as well remove the old version of maps while you're at it :-)

diegotori commented:
Re-synced branch with master. Also removed trailing version and removed android-18 as well as the Google API variant from TravisCI config.

Lemme know if you want the latter reverted if android-18 is still necessary for unit testing other Robolectric modules.
